src/main/cpp/src/get_json_object.cu

```diff
@@ -1154,6 +1154,73 @@ std::vector<std::unique_ptr<cudf::column>> get_json_object(
   return output;
 }
 
+std::vector<std::unique_ptr<cudf::column>> get_json_object(
+  cudf::strings_column_view const& input,
+  std::vector<std::vector<std::tuple<path_instruction_type, std::string, int32_t>>> const&
+    json_paths,
+  int64_t memory_budget_bytes,
+  int32_t parallel_override,
+  rmm::cuda_stream_view stream,
+  rmm::device_async_resource_ref mr)
+{
+  auto const num_outputs = json_paths.size();
+
+  // Input is empty or all nulls - just return all null columns.
+  if (input.is_empty() || input.size() == input.null_count()) {
+    std::vector<std::unique_ptr<cudf::column>> output;
+    for (std::size_t idx = 0; idx < num_outputs; ++idx) {
+      output.emplace_back(std::make_unique<cudf::column>(input.parent(), stream, mr));
+    }
+    return output;
+  }
+
+  std::vector<
+    std::tuple<std::vector<std::tuple<path_instruction_type, std::string, int32_t>>, std::size_t>>
+    sorted_paths;
+  for (std::size_t i = 0; i < json_paths.size(); i++) {
+    sorted_paths.emplace_back(json_paths[i], i);
+  }
+  std::sort(sorted_paths.begin(), sorted_paths.end());
+
+  auto const in_offsets =
+    cudf::detail::offsetalator_factory::make_input_iterator(input.offsets(), input.offset());
+  auto const scratch_size = calc_scratch_size(input, in_offsets, stream, mr);
+  if (memory_budget_bytes <= 0 && parallel_override <= 0) {
+    parallel_override = static_cast<int>(sorted_paths.size());
+  }
+  std::vector<std::unique_ptr<cudf::column>> output(num_outputs);
+  std::size_t starting_path = 0;
+  while (starting_path < num_outputs) {
+    std::size_t at = starting_path;
+    std::vector<std::vector<std::tuple<path_instruction_type, std::string, int32_t>>> batch;
+    std::vector<std::size_t> output_ids;
+    if (parallel_override > 0) {
+      int count = 0;
+      while (at < num_outputs && count < parallel_override) {
+        batch.push_back(std::get<0>(sorted_paths[at]));
```

Comment:
This is copying the entire path (a vector of tuple) which is expensive. It is much better to construct a `batch` as `std::vector<cudf::host_span>` so we won't have any copy.

Reply:
This requires changing line 1021-1022 above into
```
std::vector<cudf::host_span<std::tuple<path_instruction_type, std::string, int32_t> const>> const&
    json_paths,
```

Then, `batch` type should be:
```
std::vector<cudf::host_span<std::tuple<path_instruction_type, std::string, int32_t> const>> batch;
```